acme/dns_challenge.go

```diff
@@ -144,7 +145,8 @@ func checkAuthoritativeNss(fqdn, value string, nameservers []string) (bool, erro
 	return true, nil
 }
 
-// dnsQuery sends a DNS query to the given nameserver.
+// dnsQuery will attempt 3 times, a query to the given nameserver.
+// If the given nameserver is `RecursiveNameserver`, it will attempt the `altRecursiveNameserver` on the 2nd and 3rd attempts.
 // The nameserver should include a port, to facilitate testing where we talk to a mock dns server.
 func dnsQuery(fqdn string, rtype uint16, nameserver string, recursive bool) (in *dns.Msg, err error) {
```

Comment:
I considered it as an option. In the end for the sake of simplicity (and less changes)  I just went ahead as I did. I didn't think the proper alternation was worth it (it's just neater?). 

The slice 🍕 does add more flexibility when calling `FindZoneByFqdn()` which is nice, but almost all the providers I think use the default nameserver anyway?

I guess, my final opinion is "This is a better way to do it, and therefore we should" - even if my short-sightedness doesn't necessarily appreciate its application ;) 

In making this change, how about also adding either a variable retry, or set it to len(nameservers)+1?

And it'll be a couple of days before I can take another proper look at this.